src/AspNetCoreModuleV2/CommonLib/PollingAppOfflineApplication.cpp

```diff
@@ -7,25 +7,16 @@
 #include "SRWExclusiveLock.h"
 #include "HandleWrapper.h"
 
-HRESULT PollingAppOfflineApplication::CreateHandler(IHttpContext* pHttpContext, IREQUEST_HANDLER** pRequestHandler)
-{
-    try
-    {
-        *pRequestHandler = new PollingAppOfflineHandler(pHttpContext, m_strAppOfflineContent);
-    }
-    CATCH_RETURN();
-
-    return S_OK;
-}
-
 APPLICATION_STATUS PollingAppOfflineApplication::QueryStatus()
 {
     if (AppOfflineExists())
     {
-        return APPLICATION_STATUS::RUNNING;
+        return m_mode == StopWhenRemoved ? APPLICATION_STATUS::RUNNING : APPLICATION_STATUS::RECYCLED;
```

Comment:
I think a cleaner design would be to remove `m_mode` from `PollingAppOfflineApplication`, and instead have the subclasses `AppOfflineApplication` and `ServerErrorApplication` implement this logic.  It took me a while to understand how the `m_mode` was being used by the different subclasses.